pkg/auth/token_provider.go

```diff
@@ -63,5 +73,15 @@ func (c *OIDCTokenProvider) GetJWT(serviceAccount types.NamespacedName, audience
 		return "", fmt.Errorf("could not request a token for %s: %w", serviceAccount, err)
 	}
 
+	// we need a duration until this token expires, use the expiry time - (now + 30s)
+	// this gives us a buffer so that it doesn't expire between when we retrieve it and when we use it
+	expiryTtl := tokenRequestResponse.Status.ExpirationTimestamp.Time.Sub(time.Now().Add(expirationBufferTime))
+
+	c.tokenCache.Set(cacheKey(serviceAccount, audience), tokenRequestResponse.Status.Token, expiryTtl)
+
 	return tokenRequestResponse.Status.Token, nil
 }
+
+func cacheKey(serviceAccount types.NamespacedName, audience string) string {
+	return fmt.Sprintf("%s.%s.%s", serviceAccount.Namespace, serviceAccount.Name, audience)
```

Comment:
Can we use `/` as a delimiter, as a name can contain dots too (and thus the audience)? This should not really be an issue, but just to be safe here?